src/expr/src/scalar/func.rs

```diff
@@ -1483,6 +1483,28 @@ fn rescale_apd<'a>(a: Datum<'a>, scale: u8) -> Result<Datum<'a>, EvalError> {
     Ok(Datum::APD(d))
 }
 
+fn pg_column_size<'a>(a: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let sz = match a {
+        Datum::List(a) => {
+            // For record types (which can be full table rows), include the row header size
+            // since in that case users are trying to figure out how much memory the entire
+            // row is taking.
+            //
+            // Using datums_size instead of a.data().len() here is safer because it will
+            // return the size of the datams if they were packed into a Row. Although
+            // a.data().len() happens to give the correct answer (and is faster), data()
+            // is documented as for debugging only.
+            std::mem::size_of::<Row>() + repr::datums_size(a.iter())
+        }
```

Comment:
This seems fishy to me. This is going to include the row header for values of type `text list`, and also values of type `json` that are arrays. Even for records, the policy seems like it might be confusing: what if you want to know the size of a record on its own because you're storing that record in a column, rather than a record which represents an actual row.

I think maybe we should have another function `mz_row_size` which only accepts record types (enforced in the SQL type system) and tacks on the row overhead?

Reply:
Ergh I forgot about json arrays. What's an example of a text list? Could we have this function be smart and detect a ScalarType::Record? pg's function does special stuff for records, so it feels like it'd fit.

Reply:
Nevermind, yeah, just another function. Incoming.

Reply:
(TVFs return tables, not take tables as input. We just want a normal scalar function that goes from `ScalarType::Record -> i32`.)

Reply:
They are the only functions we have at the moment that take multiple columns as input, is why I mentioned it.